Sources/InteroperabilityTests/TestProvider.swift

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright 2024, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+import GRPCCore
+
+@available(macOS 13.0, *)
+public struct TestProvider: Grpc_Testing_TestService.ServiceProtocol {
```

Comment:
Let's call this `TestService` since it implements the `Test` service. ("Provider" was a term used in v1, we shouldn't use it here.)